include/boost/geometry/iterators/detail/segment_iterator/range_segment_iterator.hpp

```diff
@@ -100,23 +100,40 @@ class range_segment_iterator
             Reference
         >
 {
+private:
+    static bool has_less_than_two_elements(Range const& r)
+    {
+        iterator_type first = range_iterator_begin<Range>::apply(r);
+        iterator_type last = range_iterator_end<Range>::apply(r);
+
+        return first == last || ++first == last;
+    }
+
 public:
     typedef typename range_iterator_type<Range>::type iterator_type;
 
     // default constructor
     range_segment_iterator()
-        : m_it()
+        : m_it(), m_has_less_than_two_elements()
```

Comment:
Exactly, this is also my point, it's even semantically more clear to initialize coordinates explicitly passing a value. But currently in many places value-initialization is used, so my question is why it was choosen? For me it adds unnecessary requirement for a coordinate type and is less clear. But maybe I'm missing something?

In case my reasoning was correct, I propose to not use zero-initialization and initialize objects with constants explicitly when a representation of a specific scalar/numeric value is needed. This is valid not only for coordinate types.